cpp/src/strings/json/json_path.cu

```diff
@@ -570,9 +570,10 @@ class path_state : private parser {
               op.type          = path_operator_type::CHILD;
               op.expected_type = OBJECT;
             } else {
-              op.type  = path_operator_type::CHILD_INDEX;
-              op.index = cudf::io::parse_numeric<int>(
-                op.name.data(), op.name.data() + op.name.size_bytes(), json_opts, -1);
+              op.type          = path_operator_type::CHILD_INDEX;
+              auto const value = cudf::io::parse_numeric<int>(
+                op.name.data(), op.name.data() + op.name.size_bytes(), json_opts);
+              op.index = value.value_or(-1);
               CUDF_EXPECTS(op.index >= 0, "Invalid numeric index specified in JSONPath");
```

Comment:
Minor nit, rather than assigning and checking you could also do `CUDF_EXPECTS(value.has_value...); op.index = *value;`. Not sure which you think is cleaner.

Reply:
I made minimal change. `has_value` should be cleaner.

Reply:
what if a negative number is given in`op.name`.

Reply:
You're right, that would have been a problem. The current solution looks good.